tests/Integration/PrestaShopBundle/Translation/Provider/ThemeProviderTest.php

```diff
@@ -66,4 +149,159 @@ public function testItExtractsCatalogueFromXliffFiles()
+        ];
+
+        $emptyCatalogue = [
+            'SomeDomain' => [
+                'Foo' => '',
+                'Foo bar' => '',
+            ],
+            'SomeOtherDomain' => [
+                'Barbaz' => '',
+            ],
+        ];
+
+        return  [
+            'not empty catalogue' => [
+                false,
+                $extractedMessages,
+            ],
+            'empty catalogue' => [
+                true,
+                $emptyCatalogue,
+            ],
+        ];
+    }
+
+    public function testItLoadsCustomizedTranslationsFromDatabase()
+    {
+        $provider = new ThemeProvider(
+            $this->frontProvider,
+            $this->databaseReader,
+            $this->createMock(ThemeExtractorInterface::class),
+            $this->buildThemeRepository(),
+            $this->filesystem,
+            self::THEMES_DIR,
+            $this->themeName
+        );
+
+        // load catalogue from database translations
+        $catalogue = $provider->getUserTranslatedCatalogue('fr-FR');
+
+        $this->assertInstanceOf(MessageCatalogue::class, $catalogue);
+
+        // Check integrity of translations
+        $messages = $catalogue->all();
+        $domains = $catalogue->getDomains();
+        sort($domains);
+
+        // verify all catalogues are loaded
+        $this->assertSame([
+            'fakeDomain',
+        ], $domains);
+
+        // verify that the catalogues are complete
+        $this->assertCount(1, $messages['fakeDomain']);
+
+        // verify translations
+        $this->assertSame('Un texte inventé', $catalogue->get('Some made up text', 'fakeDomain'));
+    }
+
+    /**
+     * @return ThemeRepository
+     */
+    private function buildThemeRepository()
+    {
+        $configuration = $this->createMock(Configuration::class);
+
+        $configuration
+            ->method('get')
+            ->willReturnCallback(function ($param) {
+                $configs = [
+                    '_PS_ALL_THEMES_DIR_' => self::THEMES_DIR,
+                    '_PS_CONFIG_DIR_' => $this->configDir,
+                ];
+
+                return isset($configs[$param]) ? $configs[$param] : null;
+            });
+
+        $shop = $this->container->get('prestashop.adapter.legacy.context')->getContext()->shop;
+
+        return new ThemeRepository($configuration, $this->container->get('filesystem'), $shop);
+    }
+
+    /**
+     * @param array $messages
+     *
+     * @return MessageCatalogue
+     */
+    private function buildCatalogueFromMessages(array $messages)
+    {
+        $catalogue = new MessageCatalogue(ThemeExtractorInterface::DEFAULT_LOCALE);
+        foreach ($messages as $domain => $domainMessages) {
+            $catalogue->add($domainMessages, $domain);
+        }
+
+        return $catalogue;
+    }
+
+    /**
+     * @return string
+     */
+    private function getDefaultTranslationsDirectory()
+    {
+        return __DIR__ . '/../../../../Resources/translations';
+    }
```

Comment:
Maybe use a constant to be consistant with the `const THEMES_DIR` you already have?